drivers/hmis/spec/requests/hmis/project_services_spec.rb

```diff
@@ -0,0 +1,135 @@
+###
+
+require 'rails_helper'
+require_relative 'login_and_permissions'
+require_relative '../../support/hmis_base_setup'
+
+RSpec.describe Hmis::GraphqlController, type: :request do
+  include_context 'hmis base setup'
+  include_context 'hmis service setup'
+  let!(:access_control) { create_access_control(hmis_user, p1) }
+
+  describe 'project services query' do
+    before(:each) do
+      hmis_login(user)
+    end
+
+    let(:query) do
+      <<~GRAPHQL
+        query GetProjectServices($id: ID!, $filters: ServicesForProjectFilterOptions, $sortOrder: ServiceSortOption, $limit: Int = 10, $offset: Int = 0) {
+          project(id: $id) {
+            id
+            services(
+              limit: $limit
+              offset: $offset
+              sortOrder: $sortOrder
+              filters: $filters
+            ) {
+              offset
+              limit
+              nodesCount
+              nodes {
+                ...ServiceBasicFields
+                enrollment {
+                  id
+                  entryDate
+                  exitDate
+                  client {
+                    ...ClientNameDobVet
+                    __typename
+                  }
+                  __typename
+                }
+                __typename
+              }
+              __typename
+            }
+            __typename
+          }
+        }
+
+        fragment ServiceBasicFields on Service {
+          id
+          dateProvided
+          serviceType {
+            ...ServiceTypeFields
+            __typename
+          }
+          __typename
+        }
+
+        fragment ServiceTypeFields on ServiceType {
+          id
+          name
+          hud
+          hudRecordType
+          hudTypeProvided
+          category
+          dateCreated
+          dateUpdated
+          supportsBulkAssignment
+          __typename
+        }
+
+        fragment ClientNameDobVet on Client {
+          ...ClientName
+          dob
+          veteranStatus
+          __typename
+        }
+
+        fragment ClientName on Client {
+          id
+          lockVersion
+          firstName
+          middleName
+          lastName
+          nameSuffix
+          __typename
+        }
+      GRAPHQL
+    end
+
+    describe 'with more than a page of data' do
+      before(:each) do
+        # 30 enrollments, each with 10 services (5 HUD, 5 Custom) (300 services total)
+        30.times do |index|
+          # create WIP or non-WIP Enrollment
+          enrollment_factory = index.even? ? :hmis_hud_enrollment : :hmis_hud_wip_enrollment
+          en = create(enrollment_factory, data_source: ds1, project: p1)
+
+          # create both HUD and Custom services
+          5.times { create(:hmis_hud_service, date_provided: Date.yesterday, data_source: ds1, enrollment: en, client: en.client) }
+          5.times { create(:hmis_custom_service, date_provided: Date.yesterday, data_source: ds1, enrollment: en, client: en.client) }
```

Comment:
This is fine as it is. In the future I suggest preferring memoized variables for date/time in tests. This reduces the chance that tests will fail if the current date changes during the run. for example:

```ruby
let(:yesterday) { Date.yesterday }
```

Reply:
I'll keep that in mind! In this case it shouldn't fail if the date is different. I just set them to be the same so that the initial 50-record page has a variety of clients and enrollments.